src/robotFactory.js

```diff
@@ -1,13 +1,94 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const {
+      x = 0,
+      y = 0,
+    } = coords;
 
-class FlyingRobot {}
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class DeliveryDrone {}
+    this.coords = {
+      x: x,
+      y: y,
+    };
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name},` + ` Chip version: ${this.chipVersion},`
+    + ` Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords = {}, chipVersion) {
+    const {
+      x = 0,
+      y = 0,
+      z = 0,
+    } = coords;
+
+    super(name, weight, chipVersion);
+
+    this.chipVersion = chipVersion;
+
+    this.coords = {
+      x: x,
+      y: y,
+      z: z,
+    };
```

Comment:
No need to reset all the coords. You can add only new one
```suggestion
    this.coords.z = z;
```